docs/validator.md

```diff
@@ -0,0 +1,202 @@
+validate data and report errors whenever a component's model changes.
+
+# How does it work?
+
+This package provides client and server-side components that use [AJV](https://github.com/epoberezkin/ajv) to validate
+supplied JSON data against a JSON Schema
+
+The main work of the validator component is performed by its `validate(schemaKey, JSON)` invoker.  Here is an example
+of how the validator might be used from within node:
+
+    var fluid = require("infusion"");
+    var gpii  = fluid.registerNamespace("gpii");
+
+    require("gpii-json-schema");
+    var validator = gpii.schema.validator({
+        schemaContents: {
+          sample: { "type": "object", "properties": { "required": { "type": "boolean" } }, "required": ["required"]}
+        }
+    });
+
+    var errors = validator.validate("sample", { foo: "bar" });
+    if (errors) {
+        // Complain
+    }
+    else {
+        // Rejoice
+    }
+
+
+If there are no validation errors, we return `undefined`.  If there are validation errors, the raw output generated by
+AJV is transformed using the static function `sanitizeValidationErrors` (see below).
+
+This component is configured to work together with schemas that have been dereferenced, i. e. where all references
+to remote definitions have been reduced to the underlying JSON Schema rules.  See [the parser documentation](parser.md)
+for details.
+
+On the server-side, this is handled directly by a `gpii.schema.parser` instance.  On the client side, the
+validator works with content retrieved from [a `gpii.schema.inline.router` instance](router.md), which is already
+dereferenced.
+
+# Components
+
+## gpii.schema.validator.ajv
+
+The common grade, designed for use on the client and server-side.
+
+### Component Options
+
+The following component configuration options are supported:
+
+| Option             | Type     | Description |
+| ------------------ | -------- | ----------- |
+| `validatorOptions` | `Object` | Options to be passed to our AJV validator instance (check [the AJV options documentation](https://github.com/epoberezkin/ajv#options) for the syntax and available options). |
+
+
+### Invokers
+
+#### `{validator}.validate(key, content)`
+
+* `key {String}`:  The schema key we are validating against.
+* `content {Object}`: The JSON content we are validating.
+* Returns: An `{Object}` describing the errors, or `undefined` if there are no errors.
+
+The raw output provided by AJV is run through `sanitizeValidationErrors` (see below) to provide more user-friendly
+feedback where possible.
+
+### Static Functions
+
+
+#### `gpii.schema.validator.ajv.sanitizeValidationErrors(schemaKey, errors)`
+
+* `schemaKey {String}`: The schema key we are validating against.
+* `errors`: A map of existing error results to be sanitized (see below for examples).
+
+AJV gives us output like:
+
+```
+ [
+     {
+         "keyword": "minLength",
+         "schemaPath": "/password/minLength",
+         "dataPath": ".password",
```

Comment:
You should quote or reference whatever the AJV docs say about escaping rules here - or reconstruct it if they have no docs